src/evm/transaction-count.ts

```diff
@@ -0,0 +1,38 @@
+import * as node from '@api3/airnode-node';
+import { ethers } from 'ethers';
+import isNil from 'lodash/isNil';
+import { SponsorWalletTransactionCount } from '../types';
+import { retryGo } from '../utils';
+import { deriveSponsorWallet } from '../wallet';
+
+export const getSponsorWalletAndTransactionCount = async (
+  airnodeWallet: ethers.Wallet,
+  provider: ethers.providers.Provider,
+  currentBlock: number,
+  sponsor: string
+): Promise<node.LogsData<SponsorWalletTransactionCount | null>> => {
+  // Derive sponsorWallet address
+  // TODO: switch to node.evm.deriveSponsorWallet when @api3/airnode-node allows setting the `protocolId`
+  const sponsorWallet = deriveSponsorWallet(
+    airnodeWallet.mnemonic.phrase,
+    sponsor,
+    '2' // TODO: should this be in a centralized enum somewhere (api3/airnode-protocol maybe)?
```

Comment:
What is this number even for? It's not obvious at all so a descriptive variable name would help a lot

Reply:
FYI It's the protocol version :)

Reply:
This is the protocol ID part of the derivation path. We use 1 for RRP, 2 for PSP and 12345 for old Airkeeper beacon updater lambda. 

More info on sponsor wallet derivation path scheme: https://github.com/api3dao/airnode/blob/master/packages/airnode-admin/src/implementation.ts#L95-L109